go-controller/pkg/node/gateway_udn.go

```diff
@@ -248,8 +266,24 @@ func (udng *UserDefinedNetworkGateway) AddNetwork() error {
 // used by this UDN on the gateway side
 func (udng *UserDefinedNetworkGateway) DelNetwork() error {
 	vrfDeviceName := util.GetVRFDeviceNameForUDN(udng.networkID)
-	err := udng.vrfManager.DeleteVRF(vrfDeviceName)
+	// delete the iprules for this network
+	mgmtPortLinkName := util.GetNetworkScopedK8sMgmtHostIntfName(uint(udng.networkID))
+	mplink, err := util.GetNetLinkOps().LinkByName(mgmtPortLinkName)
+	if err != nil {
+		return fmt.Errorf("unable to get link for network %s, err: %v", udng.GetNetworkName(), err)
+	}
+	vrfTableId := util.CalculateRouteTableID(mplink.Attrs().Index)
```

Comment:
that's a good point and I remember having discussions around this when we started doing this logic in https://github.com/ovn-org/ovn-kubernetes/pull/4491

I cannot recollect the huddle discussions exactly... but we had started off trying to use networkID as means to calculate the VRFTableID but then also realised that EIPs also uses this and hence there are chances of potential conflicts when using networkID. The way EIP derives this was from ifIndexes and hence that was the only way to guarantee uniqueness when calculating VRF Table IDs...

@pperiyasamy / @martinkennelly do either of you remember other reasons for why we didn't go that way? I think it was the conflict reason.

Reply:
As for the delete error in case the mplink is gone..for now the deleteVRF/deleteUDNManagementPort is always called after the iprules are removed but in case other reasons remove the link prior itself we might be a bit screwed... maybe as part of delete I will change the logic to rely on getting the VRFIPRules using masqueradeIPs of this network instead of the VRFTableID... Will that address your concern?

Reply:
see if the latest push makes sense..

Reply:
@tssurya yes, not collide with EIP multi NIC tables reason.